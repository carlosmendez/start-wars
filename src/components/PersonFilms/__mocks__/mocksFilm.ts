export const mockFilms = [
  {
    "title": "mockFilm1",
    "releaseDate": "mockReleaseDate1",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 40
        },
        {
          "surfaceWater": 8
        }
      ]
    },
    "producers": [
      "Gary Kurtz",
      "Rick McCallum"
    ]
  },
  {
    "title": "mockFilm2",
    "releaseDate": "mockReleaseDate2",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 100
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": 10
        }
      ]
    },
    "producers": [
      "Gary Kurtz",
      "Rick McCallum"
    ]
  },
  {
    "title": "mockFilm3",
    "releaseDate": "mockReleaseDate3",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 12
        },
        {
          "surfaceWater": null
        }
      ]
    },
    "producers": [
      "Howard G. Kazanjian",
      "George Lucas",
      "Rick McCallum"
    ]
  },
  {
    "title": "4",
    "releaseDate": "mockReleaseDate4",
    "planetConnection": {
      "planets": [
        {
          "surfaceWater": 1
        },
        {
          "surfaceWater": 40
        },
        {
          "surfaceWater": 8
        },
        {
          "surfaceWater": 12
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": 0.9
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": 60
        },
        {
          "surfaceWater": 0
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        },
        {
          "surfaceWater": null
        }
      ]
    },
    "producers": [
      "Rick McCallum"
    ]
  }
];
