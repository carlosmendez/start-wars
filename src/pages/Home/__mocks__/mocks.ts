export const mockPeople = {
  "allPeople": {
    "edges": [
      {
        "node": {
          "name": "Luke Skywalkexxr",
          "id": "cGVvcGxlOjE=",
          "species": 50
        }
      },
      {
        "node": {
          "name": "C-3PO",
          "id": "cGVvcGxlOjI=",
          "species": {
            "averageHeight": null
          }
        }
      }
    ]
  }
};
