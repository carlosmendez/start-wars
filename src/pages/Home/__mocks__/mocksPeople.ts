export const mockPeople = {
  "allPeople": {
    "edges": [
      {
        "node": {
          "name": "mockName1",
          "id": "mockId1",
          "species": 50
        }
      },
      {
        "node": {
          "name": "mockName2",
          "id": "mockId2",
          "species": {
            "averageHeight": null
          }
        }
      }
    ]
  }
};
