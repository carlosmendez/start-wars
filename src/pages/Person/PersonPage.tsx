import React from 'react';

const PersonPage = () => {
  return (
    <>
      <p>Person</p>
    </>
  );
};

export default PersonPage;
