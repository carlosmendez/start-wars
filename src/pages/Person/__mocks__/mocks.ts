export const mockPerson = {
  "person": {
    "name": "Luke Skywalker",
      "birthYear": "19BBY",
      "species": null,
      "filmConnection": {
      "films": [
        {
          "title": "A New Hope",
          "releaseDate": "1977-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              }
            ]
          },
          "producers": [
            "Gary Kurtz",
            "Rick McCallum"
          ]
        },
        {
          "title": "The Empire Strikes Back",
          "releaseDate": "1980-05-17",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 100
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 10
              }
            ]
          },
          "producers": [
            "Gary Kurtz",
            "Rick McCallum"
          ]
        },
        {
          "title": "Return of the Jedi",
          "releaseDate": "1983-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 12
              },
              {
                "surfaceWater": null
              }
            ]
          },
          "producers": [
            "Howard G. Kazanjian",
            "George Lucas",
            "Rick McCallum"
          ]
        },
        {
          "title": "Revenge of the Sith",
          "releaseDate": "2005-05-19",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 12
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": 0.9
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 60
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              },
              {
                "surfaceWater": null
              }
            ]
          },
          "producers": [
            "Rick McCallum"
          ]
        }
      ],
        "pageInfo": {
        "hasNextPage": false,
          "hasPreviousPage": false,
          "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
          "endCursor": "YXJyYXljb25uZWN0aW9uOjM="
      },
      "totalCount": 4
    }
  }
};
