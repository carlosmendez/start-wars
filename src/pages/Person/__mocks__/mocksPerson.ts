export const mockPerson = {
  "person": {
    "name": "mockName1",
      "birthYear": "mockBirthYear",
      "species": null,
      "filmConnection": {
      "films": [
        {
          "title": "mockFilm1",
          "releaseDate": "1977-05-25",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 1
              },
              {
                "surfaceWater": 40
              },
              {
                "surfaceWater": 8
              }
            ]
          },
          "producers": [
            "mockProducer1",
            "mockProducer2"
          ]
        },
        {
          "title": "mockFilm2",
          "releaseDate": "1980-05-17",
          "planetConnection": {
            "planets": [
              {
                "surfaceWater": 100
              },
              {
                "surfaceWater": 8
              },
              {
                "surfaceWater": 0
              },
              {
                "surfaceWater": 10
              }
            ]
          },
          "producers": [
            "mockProducer1",
            "mockProducer3"
          ]
        },
      ],
        "pageInfo": {
        "hasNextPage": false,
          "hasPreviousPage": false,
          "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
          "endCursor": "YXJyYXljb25uZWN0aW9uOjM="
      },
      "totalCount": 2
    }
  }
};
