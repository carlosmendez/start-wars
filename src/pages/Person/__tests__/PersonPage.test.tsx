import React from 'react';
import {render} from "@testing-library/react";
import { Provider, CombinedError } from 'urql';
import { never, fromValue } from 'wonka';
import {BrowserRouter as Router} from 'react-router-dom';
import PersonPage from "../PersonPage";
import { mockPerson } from '../__mocks__/mocksPerson';

describe('Person Page component', () => {
  it('It renders correctly with response Success', () => {
    const successState = {
      executeQuery: () =>
        fromValue({
          data: {
            ...mockPerson
          }
        }),
    };
    const {container} = render(
      // @ts-ignore
      <Provider value={successState}>
        <Router>
          <PersonPage />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('it is fetching with Response error', () => {
    const errorState = {
      executeQuery: () =>
        fromValue({
          error: new CombinedError({
            networkError: Error('something went wrong!'),
          }),
        }),
    };
    const {container} = render(
      // @ts-ignore
      <Provider value={errorState}>
        <PersonPage />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('it is calling fetch', () => {
    const fetchingState = {
      executeQuery: jest.fn(() => never),
    };
    const {container} = render(
      // @ts-ignore
      <Provider value={fetchingState}>
        <PersonPage />
      </Provider>
    );
    expect(container).toMatchSnapshot();
    expect(fetchingState.executeQuery).toBeCalledTimes(1);
  });

});
