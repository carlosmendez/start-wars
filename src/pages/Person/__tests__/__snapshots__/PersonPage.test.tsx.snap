// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Page component It renders correctly with response Success 1`] = `
[
  <h1
    className="css-mj9wkl"
  >
    Luke Skywalker
  </h1>,
  <p>
    <small>
      Birth year:
    </small>
     
    19BBY
  </p>,
  <p>
    List of producers the person has worked with and how many times.
  </p>,
  <p>
    Gary Kurtz
    : 
    2
  </p>,
  <p>
    George Lucas
    : 
    1
  </p>,
  <p>
    Howard G. Kazanjian
    : 
    1
  </p>,
  <p>
    Rick McCallum
    : 
    4
  </p>,
  <span
    className="css-10pwxnt"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </span>,
]
`;

exports[`Person Page component it is fetching with Response error 1`] = `
<p>
  [Network] something went wrong!
</p>
`;
